### DISCUSSION TITLE
Change File/Folder Icons?

### DISCUSSION THREAD

**Discussion by maribox** [Schneegans/Fly-Pie#241] [Ideas & Feature Discussions]
![Screenshot from 2022-08-17 01-28-33](https://user-images.githubusercontent.com/70728036/185002464-6c9ccdba-dfd0-4731-b159-5deb2ed2036d.png)

The current implementation only leaves me to go through all Icons and display the names of them.
For the folders: Couldn't the names or at least the first letter/few letters of the folder be shown in the icon?
For the files: In Nautilus, PDF Files can have the first page as their logo, couldn't Fly-Pie do the same?

Hope I'm not missing any old Issues/Feature Requests here :D

**Comment by Schneegans** [comment]
Hi there; thanks for the suggestion! There's this long-standing related issue: https://github.com/Schneegans/Fly-Pie/issues/31 The main issue is that the solution must work with any icon theme a user may be using (e.g. an icon theme like [this](https://www.ubuntupit.com/wp-content/uploads/2018/08/Aqua-GTK-Theme-2.jpg)) and it should look good with all Fly-Pie themes (be it dark or light, or even transparent)...

If you have a good idea, feel free to contribute to the issue!
